src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -73,27 +76,27 @@ public synchronized void write(byte[] data) throws IOException {
     }
 
     public void openFile() throws IOException {
-        GZIPFileDescriptor fileDescriptor = new GZIPFileDescriptor();
+        FileDescriptor fileDescriptor = new FileDescriptor();
 
         File folder = new File(basePath);
         if (!folder.exists() && !folder.mkdirs()) {
             throw new IOException(String.format("Failed to create new directory %s", folder.getPath()));
         }
 
-        String filePath = getFilePath.get() + ".gz";
+        String filePath = getFilePath.get() + (compressionType == CompressionType.zip ? ".zip" : ".gz");
         fileDescriptor.path = filePath;
 
         File file = new File(filePath);
 
         file.createNewFile();
+
         FileOutputStream fos = new FileOutputStream(file);
         fos.getChannel().truncate(0);
 
         fileStream = new CountingOutputStream(fos);
-        gzipStream = new GZIPOutputStream(fileStream);
+        outputStream = compressionType == null ? new GZIPOutputStream(fileStream) : fileStream;
```

Comment:
shouldn't the condition be the other way around? you create a gzip stream when there is no compressiontype? that doesn't make sense

Reply:
Daniel is correct.
Please make sure you have the logic straight

Reply:
the logic is right and it works well
after talking to daniel i understand the confusion made from the naming here

Reply:
will change compression and compressionType to eventDataCompression as in eventhub